src/main/java/com/fnp/GOF/AbstractFactory/Main.java

```diff
@@ -0,0 +1,59 @@
+//////AFDP
+
+interface Engine{	
+	public void fireUp();
+}
+
+class Jet implements Engine{
+	public void fireUp(){
+		System.out.println("Jet engines On");
+	}	
+}
+
+class Helicopter implements Engine{
+	public void fireUp(){
+		System.out.println("Helicopter engines On");
+	}
+}
+
+abstract class AircraftType{
+	abstract Engine getEngine(String engineType);
+}
+
+class JetEngineFactory extends AircraftType{
+	public Engine getEngine(String engineType){
+			System.out.println("produce Jet "+ engineType);
+		return new Jet();		
+	}
+}
+
+class HelicopterEngineFactory extends AircraftType{
+	public Engine getEngine(String engineType){
+			System.out.println("produce Helicopter "+ engineType);
+		return new Helicopter();
+	}
+}
+
+class FlyCraft{
+	public static AircraftType getVehicle(String aircraft,int typeselector){
```

Comment:
Whats is the use of first argument(aircraft) here..and do we really need typeSelector to know which class to intantiate..is there any other way to determine that ??? please look for that approach.